### PR TITLE
Add inventory system, items and background music

### PR DESCRIPTION
**Pull request by Will-Cross1**
phase 1

### CODE REVIEW

**Comment by rowan04** [comment]
looked good on teams, but im going to test this branch myself before I merge

**Comment by Will-Cross1** [comment]
WOOO IT WORKS

**Comment by rowan04** [comment]
ok so a pretty serious problem - if you die when time is frozen, or just reset the game, everything is still frozen when you restart

**Comment by rowan04** [comment]
> ok so a pretty serious problem - if you die when time is frozen, or just reset the game, everything is still frozen when you restart

don't worry, i have now fixed this :)

File: maze-game/MyWorld.java [hunk 1]
```diff
@@ -54,8 +58,11 @@ public void act()
         // display timer if startTimer is true
```
Comment (rowan04) [inline maze-game/MyWorld.java:58]:
```suggestion
        // display timer if startTimer is true and stop is false
```

File: maze-game/MyWorld.java [hunk 2]
```diff
@@ -153,62 +160,118 @@ private void addGhost4()
         Ghost4.turn(Greenfoot.getRandomNumber(360));
     }
     
-    /**
-     * add the wall breaker
-     */
-    private void addWallBreaker()
+    //adds coordinate decoding for random numbers
```
Comment (rowan04) [inline maze-game/MyWorld.java:163]:
```suggestion
    // adds coordinate decoding for random numbers
```

File: maze-game/MyWorld.java [hunk 3]
```diff
@@ -153,62 +160,118 @@ private void addGhost4()
...
+    
+    
+    /**
+     * add the wall breaker
+     */
+    private void addWallBreaker()
+    {
+        // the wall breaker will spawn at one of the selected spawn points, at random
+        int[] ex = new int[1];
+        //ex must have different numbers
```
Comment (rowan04) [inline maze-game/MyWorld.java:262]:
```suggestion
        // ex must have different numbers
```

File: maze-game/MyWorld.java [hunk 4]
```diff
@@ -153,62 +160,118 @@ private void addGhost4()
...
+    public int getRandomWithExclusion(Random rnd, int start, int end, int[] exclude) 
+    {
+        int random = start + rnd.nextInt(end - start + 1 - exclude.length);
+        for (int ex : exclude) {
+            if (random < ex) {
+                break;
+            }
+            random++;
         }
+        return random;//somehow they stay on seperate sides FIX THIS
```
Comment (rowan04) [inline maze-game/MyWorld.java:251]: has this been fixed? does it need fixing?

File: maze-game/MyWorld.java [hunk 5]
```diff
@@ -153,62 +160,118 @@ private void addGhost4()
...
+        }
+        if (spawn_point == 12)
+        {
+            result[0] = 1125;
+            result[1] = 475;
+            return(result);
+        }
+        return(result);
+    }
+    //random number generator that excludes numbers
```
Comment (rowan04) [inline maze-game/MyWorld.java:241]:
```suggestion
    // random number generator that excludes numbers, so items can't spawn in the same place
```

File: maze-game/Player.java [hunk 6]
```diff
@@ -116,6 +132,33 @@ public void act()
             touchingBreakable(hasWallBreaker, dx, dy);
         }
     }
+    private void zaWurudo()
```
Comment (rowan04) [inline maze-game/Player.java:135]: I don't get and don't like this function name

File: maze-game/Player.java [hunk 7]
```diff
@@ -7,6 +7,7 @@
 public class Player extends Actor
 {
     public static boolean hasWallBreaker = false;
+    public static boolean freeze = false;
```
Comment (rowan04) [inline maze-game/Player.java:10]: why is this boolean variable needed? why can't we use the same one thats declared in MyWorld?

File: maze-game/TimeFilter.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+ * 
+ * @author (your name) 
+ * @version (a version number or a date)
+ */
+public class TimeFilter extends Actor
+{
+    private int timer;
+    public TimeFilter()   //constructor to resize intro text to be a better fit for the maze
+    {
+        GreenfootImage intro = getImage();
```
Comment (rowan04) [inline maze-game/TimeFilter.java:14]: intro?

File: maze-game/TimeFilter.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Write a description of class TimeFilter here.
+ * 
+ * @author (your name) 
+ * @version (a version number or a date)
+ */
+public class TimeFilter extends Actor
+{
+    private int timer;
+    public TimeFilter()   //constructor to resize intro text to be a better fit for the maze
```
Comment (rowan04) [inline maze-game/TimeFilter.java:12]: intro text?

File: maze-game/TimeFilter.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        intro.setTransparency(80);
+    }
+    /**
+     * Act - do whatever the TimeFilter wants to do. This method is called whenever
+     * the 'Act' or 'Run' button gets pressed in the environment.
+     */
+    public void act()
+    {
+        timer ++;
+        if(timer<=500)
```
Comment (rowan04) [inline maze-game/TimeFilter.java:25]: When i tested, this seemed like maybe a little too long. thoughts?

File: maze-game/TimeFilter.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+            MyWorld.stop = true;
+            //stop
+        }
+            
+        if(timer>500)
+        {
+            MyWorld.stop = false;
+            
+            getWorld().removeObjects(getWorld().getObjects(TimeFilter.class));
+            //start
```
Comment (rowan04) [inline maze-game/TimeFilter.java:36]: why is this comment here?

```suggestion

```

File: maze-game/TimePotion.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+ */
+public class TimePotion extends Actor
+{
+    /**
+     * Act - do whatever the TimePotion wants to do. This method is called whenever
+     * the 'Act' or 'Run' button gets pressed in the environment.
+     */
+    public TimePotion()   //constructor to resize time freeze to be a better fit for the maze
+    {
+        GreenfootImage brick = getImage();
```
Comment (rowan04) [inline maze-game/TimePotion.java:17]: brick?

File: maze-game/Time_option_1.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+ * @author (your name) 
+ * @version (a version number or a date)
+ */
+public class Time_option_1 extends Actor
+{
+    /**
+     * Act - do whatever the Time_option_1 wants to do. This method is called whenever
+     * the 'Act' or 'Run' button gets pressed in the environment.
+     */
+    public Time_option_1()   //constructor to resize breakable wall to be a better fit for the maze
```
Comment (rowan04) [inline maze-game/Time_option_1.java:15]: is this comment correct?

File: maze-game/images/time_freeze_icon.jpeg
Comment (rowan04) [inline maze-game/images/time_freeze_icon.jpeg:1]: I don't like this picture. cant we use a pixel stopwatch or something? it doesnt fit the rest of the theme. idek what it is

**Review by rowan04 — CHANGES_REQUESTED** [review]

File: maze-game/MyWorld.java (hunk 4 above)
Comment (Will-Cross1) [reply maze-game/MyWorld.java:251]: its relatively fixed I don't think it needs too much fixing its not that noticeable if it is broken

File: maze-game/TimeFilter.java (hunk 10 above)
Comment (Will-Cross1) [reply maze-game/TimeFilter.java:25]: could shorten it if you want

File: maze-game/TimePotion.java (hunk 12 above)
Comment (Will-Cross1) [reply maze-game/TimePotion.java:17]: no idea

File: maze-game/images/time_freeze_icon.jpeg
Comment (Will-Cross1) [reply maze-game/images/time_freeze_icon.jpeg:1]: この俺だ!

could change if you want

File: maze-game/TimeFilter.java (hunk 9 above)
Comment (Will-Cross1) [reply maze-game/TimeFilter.java:12]: copy pasted code...

Comment (rowan04) [reply maze-game/TimeFilter.java:12]: well can it be fixed?

File: maze-game/Player.java (hunk 7 above)
Comment (rowan04) [reply maze-game/Player.java:10]: i guess it doesnt really matter for now, as it works

File: maze-game/Breaker_icon.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * Write a description of class Breaker_icon here.
+ * 
+ * @author (your name) 
+ * @version (a version number or a date)
+ */
+public class Breaker_icon extends Actor
+{
+    public Breaker_icon()   //constructor to resize time freeze to be a better fit for the maze
```
Comment (rowan04) [inline maze-game/Breaker_icon.java:11]:
```suggestion
    public Breaker_icon()   // constructor to resize breaker icon to be a better fit for the maze
```

**Comment by Will-Cross1** [comment]
is it ready yet?

File: maze-game/Breaker_icon.java (hunk 14 above)
Comment (rowan04) [reply maze-game/Breaker_icon.java:11]: why was this marked as resolved?

**Review by rowan04 — APPROVED** [review]
happy with these

**Pull request merged**